Compile gambit for Apple M1

I'm playing around with getting gambit to compile and run in the new Apple M1 chip.   So far I've tweaked the configuration type from `arm64-apple-darwin20.2.0` to `aarch64-apple-darwin20.2.0` to make config.sub happy.  This lets me get through the ./configure step:
```

╰─▸ ./configure --enable-smp --enable-multiple-threaded-vms
checking build system type... aarch64-apple-darwin20.2.0
checking host system type... aarch64-apple-darwin20.2.0
checking target system type... aarch64-apple-darwin20.2.0
checking for i686-apple-darwin11-gcc-4.2.1... no
checking for i686-apple-darwin10-gcc-4.2.1... no
checking for gcc... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking how to run the C preprocessor... gcc -E
BUILD_TARGETS=C
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking size of void*... 8
checking size of long... 8
checking for bool... no
checking whether gcc defines __GNUC__... yes
checking whether gcc defines __clang__... yes
checking whether gcc defines __llvm__... yes
checking whether gcc defines _MSC_VER...
checking whether gcc defines _WIN32...
checking errno.h usability... yes
checking errno.h presence... yes
checking for errno.h... yes
checking float.h usability... yes
checking float.h presence... yes
checking for float.h... yes
checking signal.h usability... yes
checking signal.h presence... yes
checking for signal.h... yes
checking stdio.h usability... yes
checking stdio.h presence... yes
checking for stdio.h... yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking time.h usability... yes
checking time.h presence... yes
checking for time.h... yes
checking for unistd.h... (cached) yes
checking pwd.h usability... yes
checking pwd.h presence... yes
checking for pwd.h... yes
checking dlfcn.h usability... yes
checking dlfcn.h presence... yes
checking for dlfcn.h... yes
checking dl.h usability... no
checking dl.h presence... no
checking for dl.h... no
checking sys/dxe.h usability... no
checking sys/dxe.h presence... no
checking for sys/dxe.h... no
checking mach-o/dyld.h usability... yes
checking mach-o/dyld.h presence... yes
checking for mach-o/dyld.h... yes
checking mach/mach_time.h usability... yes
checking mach/mach_time.h presence... yes
checking for mach/mach_time.h... yes
checking for sys/types.h... (cached) yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking sys/times.h usability... yes
checking sys/times.h presence... yes
checking for sys/times.h... yes
checking sys/timeb.h usability... yes
checking sys/timeb.h presence... yes
checking for sys/timeb.h... yes
checking sys/timers.h usability... no
checking sys/timers.h presence... no
checking for sys/timers.h... no
checking sys/resource.h usability... yes
checking sys/resource.h presence... yes
checking for sys/resource.h... yes
checking sys/wait.h usability... yes
checking sys/wait.h presence... yes
checking for sys/wait.h... yes
checking sys/mman.h usability... yes
checking sys/mman.h presence... yes
checking for sys/mman.h... yes
checking syslog.h usability... yes
checking syslog.h presence... yes
checking for syslog.h... yes
checking execinfo.h usability... yes
checking execinfo.h presence... yes
checking for execinfo.h... yes
checking netdb.h usability... yes
checking netdb.h presence... yes
checking for netdb.h... yes
checking grp.h usability... yes
checking grp.h presence... yes
checking for grp.h... yes
checking fpu_control.h usability... no
checking fpu_control.h presence... no
checking for fpu_control.h... no
checking os2.h usability... no
checking os2.h presence... no
checking for os2.h... no
checking dos.h usability... no
checking dos.h presence... no
checking for dos.h... no
checking direct.h usability... no
checking direct.h presence... no
checking for direct.h... no
checking wdefwin.h usability... no
checking wdefwin.h presence... no
checking for wdefwin.h... no
checking tfork.h usability... no
checking tfork.h presence... no
checking for tfork.h... no
checking curses.h usability... yes
checking curses.h presence... yes
checking for curses.h... yes
checking ncurses.h usability... yes
checking ncurses.h presence... yes
checking for ncurses.h... yes
checking netinet/in.h usability... yes
checking netinet/in.h presence... yes
checking for netinet/in.h... yes
checking arpa/inet.h usability... yes
checking arpa/inet.h presence... yes
checking for arpa/inet.h... yes
checking termios.h usability... yes
checking termios.h presence... yes
checking for termios.h... yes
checking pty.h usability... no
checking pty.h presence... no
checking for pty.h... no
checking stropts.h usability... no
checking stropts.h presence... no
checking for stropts.h... no
checking libutil.h usability... no
checking libutil.h presence... no
checking for libutil.h... no
checking util.h usability... yes
checking util.h presence... yes
checking for util.h... yes
checking sys/fpu.h usability... no
checking sys/fpu.h presence... no
checking for sys/fpu.h... no
checking fenv.h usability... yes
checking fenv.h presence... yes
checking for fenv.h... yes
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking sys/ioctl.h usability... yes
checking sys/ioctl.h presence... yes
checking for sys/ioctl.h... yes
checking sys/socket.h usability... yes
checking sys/socket.h presence... yes
checking for sys/socket.h... yes
checking for strings.h... (cached) yes
checking for memory.h... (cached) yes
checking sys/sysctl.h usability... yes
checking sys/sysctl.h presence... yes
checking for sys/sysctl.h... yes
checking sys/syscall.h usability... yes
checking sys/syscall.h presence... yes
checking for sys/syscall.h... yes
checking linux/fs.h usability... no
checking linux/fs.h presence... no
checking for linux/fs.h... no
checking ws2tcpip.h usability... no
checking ws2tcpip.h presence... no
checking for ws2tcpip.h... no
checking TargetConditionals.h usability... yes
checking TargetConditionals.h presence... yes
checking for TargetConditionals.h... yes
checking AvailabilityMacros.h usability... yes
checking AvailabilityMacros.h presence... yes
checking for AvailabilityMacros.h... yes
checking poll.h usability... yes
checking poll.h presence... yes
checking for poll.h... yes
checking for environ... yes
checking crt_externs.h usability... yes
checking crt_externs.h presence... yes
checking for crt_externs.h... yes
checking for _NSGetEnviron... yes
checking for mach-o/dyld.h... (cached) yes
checking for _NSGetExecutablePath... yes
checking for remove... yes
checking for rename... yes
checking for sys/stat.h... (cached) yes
checking for sys/types.h... (cached) yes
checking for fcntl.h... (cached) yes
checking for mkdir... yes
checking dirent.h usability... yes
checking dirent.h presence... yes
checking for dirent.h... yes
checking for opendir... yes
checking for fdopendir... yes
checking stat.h usability... no
checking stat.h presence... no
checking for stat.h... no
checking for sys/stat.h... (cached) yes
checking for stat... yes
checking for stat64... yes
checking for struct stat64... no
checking for getcwd... yes
checking for sigset_t... yes
checking for socklen_t... yes
checking for library containing asin... none required
checking for library containing dlopen... none required
checking for library containing shl_load... no
checking for library containing socket... none required
checking for library containing gethostbyname... none required
checking for library containing hstrerror... none required
checking for library containing openpty... none required
checking for library containing clock_gettime... none required
checking for library containing timeBeginPeriod... no
checking for backtrace_symbols_fd... yes
checking for chdir... yes
checking for execvp... yes
checking for fcntl... yes
checking for getgrnam... yes
checking for getpid... yes
checking for getppid... yes
checking for getpwnam... yes
checking for ioctl... yes
checking for link... yes
checking for mkfifo... yes
checking for mmap... yes
checking for open... yes
checking for pipe... yes
checking for readlink... yes
checking for rmdir... yes
checking for socket... yes
checking for socketpair... yes
checking for strerror... yes
checking for symlink... yes
checking for sysconf... yes
checking for sysctl... yes
checking for sysctlbyname... yes
checking for syslog... yes
checking for unlink... yes
checking for utimes... yes
checking for waitpid... yes
checking for fstatat... yes
checking for linkat... yes
checking for mkdirat... yes
checking for mkfifoat... no
checking for openat... yes
checking for readlinkat... yes
checking for renameat... yes
checking for renameat2... no
checking for renameatx_np... yes
checking for symlinkat... yes
checking for unlinkat... yes
checking for tcgetsetattr... yes
checking for sigaction... yes
checking for sigemptyset/sigaddset... yes
checking for sigprocmask... yes
checking for signal... yes
checking for clock_gettime... yes
checking for mach_absolute_time... yes
checking for getclock... no
checking for gettimeofday... yes
checking for ftime... yes
checking for time... yes
checking for sleep... yes
checking for nanosleep... yes
checking for setitimer... yes
checking for dos_setvect... no
checking for DosStartTimer... no
checking for VInstall... no
checking for getrusage... yes
checking for times... yes
checking for clock... yes
checking for DosQuerySysInfo... no
checking for shl_load... no
checking for DosLoadModule... no
checking for dxe_load... no
checking for GetDiskFragment... no
checking for dlopen... yes
checking for NSLinkModule... yes
checking for gethostname... yes
checking for inet_pton... yes
checking for getaddrinfo... yes
checking for gethostbyname... yes
checking for gethostbyaddr... yes
checking for getservbyname... yes
checking for getservbyport... yes
checking for getprotobyname... yes
checking for getprotobynumber... yes
checking for getnetbyname... yes
checking for select... yes
checking for poll... yes
checking for ppoll... no
checking for openpty... yes
checking for getpt... no
checking for ptsname... yes
checking for ctermid... yes
checking for isastream... no
checking for hstrerror... yes
checking for snprintf... yes
checking for get_fpc_csr... no
checking for posix threads with CFLAGS=' ' and LIBS=' '... yes
checking for thread local storage class... _Thread_local
checking whether gcc accepts -no-cpp-precomp...  -no-cpp-precomp
checking whether gcc accepts -Wall...  -Wall
checking whether gcc accepts -W...  -W
checking whether gcc accepts -Wno-unused...  -Wno-unused
checking whether gcc accepts -Wno-write-strings...  -Wno-write-strings
checking whether gcc accepts -Wdisabled-optimization...  -Wdisabled-optimization
checking whether gcc accepts -g...  -g
checking whether gcc accepts -p...  -p
checking whether gcc accepts -pg...  -pg
checking whether gcc accepts -ftest-coverage...  -ftest-coverage
checking whether gcc accepts -fprofile-arcs...  -fprofile-arcs
checking whether gcc accepts -fbranch-probabilities...
checking whether gcc accepts -O1...  -O1
checking whether gcc accepts -O2...  -O2
checking whether gcc accepts -O3...  -O3
checking whether gcc accepts -Ofast...  -Ofast
checking whether gcc accepts -fno-math-errno...  -fno-math-errno
checking whether gcc accepts -fno-trapping-math...  -fno-trapping-math
checking whether gcc accepts -fno-strict-aliasing...  -fno-strict-aliasing
checking whether gcc accepts -fwrapv...  -fwrapv
checking whether gcc accepts -ftrapv...  -ftrapv
checking whether gcc accepts -fomit-frame-pointer...  -fomit-frame-pointer
checking whether gcc accepts -fno-ipa-ra...
checking whether gcc accepts -fPIC...  -fPIC
checking whether gcc accepts -fpic...  -fpic
checking whether gcc accepts -fno-common...  -fno-common
checking whether gcc accepts -dynamiclib...  -dynamiclib
checking whether gcc accepts -bundle...  -bundle
checking whether gcc accepts -flat_namespace -undefined suppress...  -flat_namespace -undefined suppress
checking for X... no
checking whether ln -s works... yes
checking for ranlib... ranlib
checking for ar... ar
checking whether make sets $(MAKE)... yes
configure: creating ./config.status
config.status: creating makefile
config.status: creating include/makefile
config.status: creating include/gambit.h
config.status: creating lib/makefile
config.status: creating lib/module-common.mk
config.status: creating gsi/makefile
config.status: creating gsc/makefile
config.status: creating bin/makefile
config.status: creating bin/gambvcs.unix
config.status: creating bin/gambvcs.bat.windows
config.status: creating bin/gambuild-C.unix
config.status: creating bin/gambuild-C.bat.windows
config.status: creating bin/gambuild-js.unix
config.status: creating bin/gambuild-js.bat.windows
config.status: creating bin/gambuild-python.unix
config.status: creating bin/gambuild-python.bat.windows
config.status: creating bin/gambuild-ruby.unix
config.status: creating bin/gambuild-ruby.bat.windows
config.status: creating bin/gambuild-php.unix
config.status: creating bin/gambuild-php.bat.windows
config.status: creating bin/gambuild-java.unix
config.status: creating bin/gambuild-java.bat.windows
config.status: creating bin/gambuild-x86.unix
config.status: creating bin/gambuild-x86.bat.windows
config.status: creating bin/gambuild-x86-64.unix
config.status: creating bin/gambuild-x86-64.bat.windows
config.status: creating bin/gambuild-arm.unix
config.status: creating bin/gambuild-arm.bat.windows
config.status: creating bin/gambuild-riscv-32.unix
config.status: creating bin/gambuild-riscv-32.bat.windows
config.status: creating bin/gambuild-riscv-64.unix
config.status: creating bin/gambuild-riscv-64.bat.windows
config.status: creating bin/gambdoc.unix
config.status: creating bin/gambdoc.bat.windows
config.status: creating misc/makefile
config.status: creating doc/makefile
config.status: creating tests/makefile
config.status: creating examples/makefile
config.status: creating examples/distr-comp/makefile
config.status: creating examples/pi/makefile
config.status: creating examples/ring/makefile
config.status: creating examples/web-repl/makefile
config.status: creating examples/web-server/makefile
config.status: creating examples/tcltk/makefile
config.status: creating examples/Xlib-simple/makefile
config.status: creating examples/pthread/makefile
config.status: creating examples/misc/makefile
config.status: creating bench/makefile
config.status: creating contrib/makefile
config.status: creating contrib/GambitREPL/makefile
config.status: creating contrib/xactlog/makefile
config.status: creating prebuilt/makefile
config.status: creating prebuilt/macosx/makefile
config.status: creating prebuilt/macosx/build-phase2
config.status: creating prebuilt/windows/makefile
config.status: creating prebuilt/windows/build-phase2
config.status: creating githooks/makefile
config.status: creating include/config.h
config.status: include/config.h is unchanged
configure:
**************************************************************************
***                                                                    ***
*** The clang compiler is being used.  Please be aware that the build  ***
*** may take a very long time (over 30 minutes on a fairly fast        ***
*** machine with clang 3.0).  Configuring with GNU GCC will give a     ***
*** faster build (1 min on the same machine with GNU GCC 4.2.1) and    ***
*** the executable will be faster.                                     ***
***                                                                    ***
**************************************************************************
```

And then the make step completes successfully.

So far so good, _but_, then I get a heap overflow compiling modules:

```
 ╰─▸ make modules
>>>> building module _define-library for C
>>>> building module _define-library/define-library-expand for C
>>>> building module _define-library/debug for C
>>>> building module _digest for C
>>>> building module _geiser for C
>>>> building module _git for C
>>>> building module _hamt for C
*** FATAL ERROR -- Heap overflow
make[2]: *** [modules-pre] Error 70
make[1]: *** [_hamt] Error 1
make: *** [modules] Error 2
```

Happy to try to pin down what's going on but if anyone can give me some pointers for good places to start looking I'd appreciate it.

--- a/config.sub
+++ b/config.sub
@@ -1156,7 +1156,7 @@ case $cpu-$vendor in
 		case $cpu in
 			1750a | 580 \
 			| a29k \
-			| aarch64 | aarch64_be \
+			| arm64 | aarch64 | aarch64_be \
 			| abacus \
 			| alpha | alphaev[4-8] | alphaev56 | alphaev6[78] \
 			| alpha64 | alpha64ev[4-8] | alpha64ev56 | alpha64ev6[78] \

